qp_spades/qp_spades.py

```diff
@@ -6,12 +6,134 @@
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
 import pandas as pd
-
-from os.path import basename
+from os import environ
+from os.path import join, exists
+from glob import glob
+from subprocess import run, PIPE
 
 from qiita_client import ArtifactInfo
 from qiita_client.util import system_call
 
+# resources per job
+WALLTIME = '200:00:00'
+MAX_RUNNING = 8
+FINISH_WALLTIME = '10:00:00'
+FINISH_MEMORY = '4g'
+
+
+def spades_to_array(directory, output_dir, prefix_to_name, url,
+                    job_id, params):
+    environment = environ["ENVIRONMENT"]
+    ppn = params["threads"]
+    memory = params["memory"]
+
+    # 1. create file list
+    files = []
+    for prefix, sample_name in prefix_to_name.items():
+        fps = sorted(glob(join(directory, prefix + '*')))
+        # this should never occur but better to confirm
+        if len(fps) != 2:
+            mgs = f'There are multiple files that match "{prefix}"'
+            return False, None, mgs
+        files.append('\t'.join([fps[0], fps[1], prefix]))
+
+    # 2. format main comand
+    # cmd = (f'{spades_cmd} > {out_dir}/%s.spades.log 2>&1 && '
+    #            f'mv {out_dir}/{sample_name}/scaffolds.fasta {out_fp}' % (
+    #             fwd_fp, rev_fp, sample_name, sample_name))
+    command = (
+        f'spades.py --{params["type"]} -t {ppn} -m {memory} '
+        f'-k {params["k-mers"]} -o $OUTDIR/$SNAME')
+    if params['merging'].startswith('flash '):
+        # get read lenght quickly; note that we are going to assume
+        # that (1) the forward and reverse are the same lenght and (2)
+        # all file pairs have the same lenght so only calculate once
+        fp = glob(join(directory, list(prefix_to_name)[0] + '*'))[0]
+        std_out, std_err, return_value = system_call(
+                f'gunzip -c {fp} | head -n 2')
```

Comment:
Why the `head -n 2`?

Reply:
To just get the header (line 1) and the sequence (line 2)

Reply:
Ah okay, then suggest `zcat {fp} | head -n 2`. I *think* it's the same thing, but on a quick read it looked like the file was being uncompressed in full. `zcat` may avoid that misunderstanding. 
```suggestion
                f'zcat {fp} | head -n 2')
```